### ISSUE TITLE
HTTP Status 404 – Not Found

### ISSUE THREAD

**Issue by tall27** [itiligent/Guacamole-Install#41]
**Describe the issue:**
After successful install of the latest version, I can access the URL, but getting HTTP Status 404 – Not Found error 
HTTP Status 404 – Not Found
Type Status Report
Message The requested resource [/guacamole/] is not available
Description The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.
Apache Tomcat/10.1.16 (Ubuntu)

**Steps to reproduce the issue:**
Fresh Ubuntu on AWS. ports 80, 443 open from 0\0 port ; port 22 from my IP.
Successfully got a certificate from Let's encrypt and can access to the Nginx web site.

**Describe the software environment:**
- OS & version
uname -a :
`Linux ip-10-0-1-181 6.8.0-1008-aws #8-Ubuntu SMP Sat Apr 20 00:46:25 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
based on AWS ubuntu/images/hvm-ssd-gp3/ubuntu-noble-24.04-amd64-server-20240423 image
- Include platform i.e. physical, virtual, cloud image etc
AWS environment 
- List other applications present
nothing installed. the script run as the very first command.
- List other services or tasks the system also currently provides
- Details any other relevant background context about your OS, its location, method of management or access, firewall settings etc.

**Logs / screenshots / error outputs etc**
- Where possible, provide as much background detail as possible to help explain your problem through the outputs you have available.
![image](https://github.com/itiligent/Guacamole-Install/assets/51923785/f9e19406-77a1-403b-bbbb-970d4e5d1e73)

**Troubleshooting steps already taken?:**
I've tried to create a new EC2 instance and run it again.

**Comment by tall27** [comment]
root@ip-10-0-1-181:/home/ubuntu# **curl  http://127.0.0.1:8080/**
<?xml version="1.0" encoding="ISO-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
    <title>Apache Tomcat</title>
</head>

<body>
<h1>It works !</h1>

<p>If you're seeing this page via a web browser, it means you've setup Tomcat successfully. Congratulations!</p>

<p>This is the default Tomcat home page. It can be found on the local filesystem at: <code>/var/lib/tomcat10/webapps/ROOT/index.html</code></p>

<p>Tomcat veterans might be pleased to learn that this system instance of Tomcat is installed with <code>CATALINA_HOME</code> in <code>/usr/share/tomcat10</code> and <code>CATALINA_BASE</code> in <code>/var/lib/tomcat10</code>, following the rules from <code>/usr/share/doc/tomcat10-common/RUNNING.txt.gz</code>.</p>

<p>You might consider installing the following packages, if you haven't already done so:</p>

<p><b>tomcat10-docs</b>: This package installs a web application that allows to browse the Tomcat 10 documentation locally. Once installed, you can access it by clicking <a href="docs/">here</a>.</p>

<p><b>tomcat10-examples</b>: This package installs a web application that allows to access the Tomcat 10 Servlet and JSP examples. Once installed, you can access it by clicking <a href="examples/">here</a>.</p>

<p><b>tomcat10-admin</b>: This package installs two web applications that can help managing this Tomcat instance. Once installed, you can access the <a href="manager/html">manager webapp</a> and the <a href="host-manager/html">host-manager webapp</a>.</p>

<p>NOTE: For security reasons, using the manager webapp is restricted to users with role "manager-gui". The host-manager webapp is restricted to users with role "admin-gui". Users are defined in <code>/etc/tomcat10/tomcat-users.xml</code>.</p>

</body>
</html>

===================================================================

root@ip-10-0-1-181:/home/ubuntu# **curl  http://127.0.0.1:8080/guacamole**
<!doctype html><html lang="en"><head><title>HTTP Status 404 – Not Found</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 404 – Not Found</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> The requested resource [&#47;guacamole] is not available</p><p><b>Description</b> The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.</p><hr class="line" /><h3>Apache Tomcat/10.1.16 (Ubuntu)</h3></body></html>root@ip-10-0-1-181:/home/ubuntu#

**Comment by itiligent** [comment]
Hi,
 
The issue relates to you using Ubuntu 24.04 (released 1.5 mth ago) before I had got around to adding a workaround to support it - (as per the prerequisites section of the Readme.)

Today I have added support for Ubuntu 24.04 by adding the previous Jammy repo to allow a Tomcat 9 install. Should work fine on your latest Ubuntu AWS instance now.

Background: The need for many workaround's at present is because Guacamole requires Tomcat9 and as time goes on, more and more distros are defaulting to Tomcat10.  (The Gucamole devs dont yet have a timeline for adding support for Tomcat10, but Tomcat9 is EOL in about a year or so so this will happen sooner than later.)  I could simply apply a source build of Tomcat for all versions, but it is my preferrence to try to rely on distro based Tomcat packages for ease of updates via APT.
